crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -779,120 +817,233 @@ mod tests {
                 assert_eq!(&site_packages_directories, &[expected_venv_site_packages]);
             }
         }
+
+        fn assert_system_environment(
+            &self,
+            env: &SystemEnvironment,
+            expected_env_path: &SystemPathBuf,
+        ) {
+            assert!(
+                self.virtual_env.is_none(),
+                "`assert_system_environment` should only be used when `virtual_env` is not populated"
+            );
+
+            assert_eq!(
+                env.root_path,
+                SysPrefixPath {
+                    inner: self.system.canonicalize_path(expected_env_path).unwrap(),
+                    origin: SysPrefixPathOrigin::VirtualEnvVar,
+                }
+            );
+
+            let site_packages_directories = env.site_packages_directories(&self.system).unwrap();
+
+            let expected_site_packages = if cfg!(target_os = "windows") {
+                SystemPathBuf::from(&*format!(
+                    r"\Python3.{}\Lib\site-packages",
+                    self.minor_version
+                ))
+            } else if self.free_threaded {
+                SystemPathBuf::from(&*format!(
+                    "/Python3.{minor_version}/lib/python3.{minor_version}t/site-packages",
+                    minor_version = self.minor_version
+                ))
+            } else {
+                SystemPathBuf::from(&*format!(
+                    "/Python3.{minor_version}/lib/python3.{minor_version}/site-packages",
+                    minor_version = self.minor_version
+                ))
+            };
```

Comment:
this looks repeated from the virtual environment branch -- worth factoring it out into a helper function?

Reply:
It feels low priority to me in a test setup, but yeah we could.

Reply:
Yeah, I just don't like how these strings are pretty complicated and are duplicated in both branches -- it feels kinda fragile. But I don't feel strongly; feel free to disregard if you don't think it's worth spending time on!

Reply:
The test setup feels fragile to me, I think that's why I wasn't motivated to do more consolidation. I can look at doing so in a follow-up though.

Reply:
it _is_ fragile, yeah, agreed